SonarRunner.Shim/SonarProjectPropertiesValidator.cs

```diff
@@ -0,0 +1,40 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="SonarRunnerWrapper.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+using System;
+using System.IO;
+using System.Linq;
+using System.Collections.Generic;
+using SonarQube.Common;
+
+namespace SonarRunner.Shim
+{
+    public static class SonarProjectPropertiesValidator
+    {
+        public static void Validate(string sonarRunnerCwd, IDictionary<ProjectInfo, ProjectInfoValidity> projects, Action onValid, Action<IList<string>> onInvalid)
```

Comment:
A comment explaining what the "onInvalid" arguments represent would help (a list of the files that should be deleted?)